lexos/models/topword_model.py

```diff
@@ -0,0 +1,377 @@
+        readable_result = [
+            TopwordModel._z_test_word_list(
+                data_series_one=pd.Series(
+                    data=self._doc_term_matrix.loc[file_id],
+                    index=words,
+                    name='Document "%s"' % (self._id_temp_label_map[file_id])),
+                data_series_two=pd.Series(
+                    data=group_data.loc[class_label],
+                    index=words,
+                    name='Class "%s"' % class_label))
+            for file_id, class_label in file_class_combinations]
+
+        return readable_result
+
+    def _analyze_class_to_class(self, class_division_map: pd.DataFrame) -> \
+            ReadableResult:
+        """Detect if a given word is an anomaly.
+
+        While doing so, this method compares the occurrence of a given word
+        in a class of segments to the occurrence of the same word in other
+        class of segments.
+        :return: a list of series, each series has a readable name and the
+                 result, which contains words with corresponding z-scores.
+        """
+        # Trap possible empty input error.
+        assert not self._doc_term_matrix.empty, SEG_NON_POSITIVE_MESSAGE
+
+        # Initialize, get all words that appear at least once in whole corpus.
+        words = self._doc_term_matrix.columns.values
+
+        # Get all class labels.
+        class_labels = class_division_map.index.values
+
+        # Get all unique combinations of every two labels, so we can compare
+        # one class against other class(es).
+        label_combinations = list(itertools.combinations(class_labels, 2))
+
+        # Split DTM into groups and find word count sums of each group.
+        group_sums = [np.sum(self._doc_term_matrix.values[row], axis=0)
+                      for row in class_division_map.values]
+
+        # Put groups word count sums into a data frame.
+        group_data = \
+            pd.DataFrame(data=group_sums, index=class_labels, columns=words)
+
+        # Generate analysis result.
+        readable_result = [
+            TopwordModel._z_test_word_list(
+                data_series_one=pd.Series(
+                    data=group_data.loc[group_one_label],
+                    index=words,
+                    name='Class "%s"' % group_one_label),
+                data_series_two=pd.Series(
+                    data=group_data.loc[group_two_label],
+                    index=words,
+                    name='Class "%s"' % group_two_label))
+            for group_one_label, group_two_label in label_combinations]
+
+        return readable_result
+
+    def _get_result(self, class_division_map: pd.DataFrame) -> TopwordResult:
+        """Call the right method corresponding to user's selection.
+
+        :return: a namedtuple that holds the topword result, which contains a
+                 header and a list of pandas series."""
+
+        if self._topword_front_end_option.analysis_option == "allToPara":
```

Comment:
self._topword_front_end_option.analysis_option should be a enum:
```python
from enum import Enum
class TopwordsAnalysisType(Enum):
    ALL_TO_PARA = "allToPara"
    CLASS_TO_PARA = "classToPara"
    CLASS_TO_CLASS = "classToClass"
```
use it like this:
```python
topword_analysis_option = self._topword_front_end_option.analysis_option

if topword_analysis_option is TopwordsAnalysisType.ALL_TO_PARA:
    ...
elif topword_analysis_option is TopwordsAnalysisType.CLASS_TO_PARA:
    ...
elif topword_analysis_option is TopwordsAnalysisType.CLASS_TO_CLASS:
    ...
else:
    raise ValueError("invalid topword option")

```

https://docs.python.org/3/library/enum.html